Utils.py

```diff
@@ -1074,3 +1074,30 @@ def is_iterable_except_str(obj: object) -> TypeGuard[typing.Iterable[typing.Any]
     if isinstance(obj, str):
         return False
     return isinstance(obj, typing.Iterable)
+
+
+def get_full_typename(t: type) -> str:
+    module = t.__module__
+    if module and module != "builtins":
+        return f"{module}.{t.__qualname__}"
+    return t.__qualname__
+
+
+def get_all_causes(ex: Exception) -> str:
+    """Return a string describing the recursive causes of this exception.
```

Comment:
This text should be on the line below, and this docstring needs param and return descriptions. One above probably ought to get a docstring too

Reply:
Summary doctext being on the line below the opening quotes is inconsistent with the only other multi-paragraph doc comment in this file and the style guide seems to allow it if it's short enough.